ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -697,16 +726,30 @@ func (r *RayJobReconciler) checkK8sJobAndUpdateStatusIfNeeded(ctx context.Contex
 	for _, cond := range job.Status.Conditions {
 		if cond.Type == batchv1.JobFailed && cond.Status == corev1.ConditionTrue {
 			logger.Info("The submitter Kubernetes Job has failed. Attempting to transition the status to `Failed`.", "RayJob", rayJob.Name, "Submitter K8s Job", job.Name, "Reason", cond.Reason, "Message", cond.Message)
-			rayJob.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusFailed
 			// The submitter Job needs to wait for the user code to finish and retrieve its logs.
 			// Therefore, a failed Submitter Job indicates that the submission itself has failed or the user code has thrown an error.
 			// If the failure is due to user code, the JobStatus and Job message will be updated accordingly from the previous reconciliation.
 			if rayJob.Status.JobStatus == rayv1.JobStatusFailed {
 				rayJob.Status.Reason = rayv1.AppFailed
+
+				failedCount := int32(0)
+				if rayJob.Status.Failed != nil {
+					failedCount = *rayJob.Status.Failed
+				}
+				failedCount++
+				rayJob.Status.Failed = pointer.Int32(failedCount)
 			} else {
 				rayJob.Status.Reason = rayv1.SubmissionFailed
```

Comment:
Should we also update `failedCount` for `SubmissionFailed`? If the submitter K8s Job failed consistently due to misconfiguration and the `BackoffLimit` is larger than 1, the RayJob seems not to transition to `JobDeploymentStatusFailed` forever.

Reply:
done